version: "3.8"
services:
  jupyter:
    container_name: adsproject_jupyer
    build:
      context: .
      dockerfile: DockerfileJupyterNotebook.dockerfile
    ports:
      - "8888:8888"
    networks:
      - my-network
  
  postgres:
    image: postgres:latest
    container_name: adsproject_postgres
    environment:
      POSTGRES_DB: adsproject
      POSTGRES_USER: ads
      POSTGRES_PASSWORD: admin123!
    volumes:
      - postgres_data:/var/lib/postgresql/data
    ports:
      - "5432:5432"
    healthcheck:
      test: pg_isready -U check_connection -h localhost -p 5432
      interval: 15s
      timeout: 2s
      retries: 3
      start_period: 1s
    restart: unless-stopped
    networks:
      - my-network

  db_ops:
    container_name: ads_useless_container
    depends_on:
      postgres:
        condition: service_healthy
    build:
      context: .
      dockerfile: DockerfileDBOps.dockerfile
    networks:
      - my-network
  
  selenium:
    container_name: ads_scraper
    shm_size: '2gb'
    image: seleniarm/standalone-chromium:111.0.5563.64
    ports:
      - "4444:4444"
      - "7900:7900"
    networks:
      - my-network
  
volumes:
  postgres_data:

networks:
  my-network:
